// @forge/cli-shared/out/http-client/feedback-post-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpsFeedbackPostClient = void 0;
const tslib_1 = require("tslib");
const node_fetch_1 = tslib_1.__importDefault(require("node-fetch"));
const prepareRequestBody = (feedback) => {
    return {
        fields: [
            {
                id: 'summary',
                value: feedback.summary
            },
            {
                id: 'description',
                value: feedback.description
            },
            {
                id: 'email',
                value: feedback.email
            }
        ]
    };
};
class HttpsFeedbackPostClient {
    constructor() {
        this.forgeFeedbackCollectorUrl = 'https://jsd-widget.atlassian.com/api/embeddable/b8a921af-5e55-4b6f-a219-590251ab908f/request?requestTypeId=326';
    }
    async sendFeedback(feedback) {
        return (0, node_fetch_1.default)(this.forgeFeedbackCollectorUrl, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(prepareRequestBody(feedback))
        });
    }
}
exports.HttpsFeedbackPostClient = HttpsFeedbackPostClient;
