// @forge/cli-shared/out/apps/create-app-graphql-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAppGraphQLClient = exports.CannotCreateAppError = void 0;
const graphql_1 = require("../graphql");
const ui_1 = require("../ui");
class CannotCreateAppError extends Error {
}
exports.CannotCreateAppError = CannotCreateAppError;
class CreateAppGraphQLClient {
    constructor(graphqlClient) {
        this.graphqlClient = graphqlClient;
    }
    async createApp(input) {
        const createAppQuery = `
      mutation forge_cli_createApplication($input: CreateAppInput!) {
        createApp(input: $input) {
          success
          errors {
            message
            extensions {
              errorType
            }
          }
          app {
            id
            environments {
              key
            }
          }
        }
      }
    `;
        const { response: { createApp: { success, errors, app } }, requestId } = await this.graphqlClient.mutate(createAppQuery, {
            input
        });
        const error = (0, graphql_1.getError)(errors);
        if (!success) {
            throw new graphql_1.GraphQlMutationError(`${error.message} (requestId: ${requestId || 'unknown'})`, {
                requestId,
                code: error.code,
                statusCode: error.statusCode
            });
        }
        if (!app) {
            throw new CannotCreateAppError(ui_1.Text.create.error.app.creation);
        }
        return app;
    }
}
exports.CreateAppGraphQLClient = CreateAppGraphQLClient;
