// @forge/cli-shared/out/apps/register-app.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegisterAppCommand = void 0;
const path_1 = require("path");
const ui_1 = require("../ui");
class RegisterAppCommand {
    constructor(appClient, appConfigWriter, logger) {
        this.appClient = appClient;
        this.appConfigWriter = appConfigWriter;
        this.logger = logger;
    }
    async execute(input, shouldPrependAppName = false) {
        const result = await this.registerApp(input, '.', shouldPrependAppName);
        return {
            analytics: { appId: result.id, appName: input.name },
            environments: result.environments.map((env) => env.key)
        };
    }
    async registerApp(input, directory, shouldPrependAppName) {
        this.logger.info(ui_1.Text.create.taskRegister);
        const result = await this.appClient.createApp(input);
        const cwd = process.cwd();
        process.chdir((0, path_1.resolve)(cwd, directory));
        try {
            await this.appConfigWriter.writeConfigSection({ id: result.id });
            if (shouldPrependAppName) {
                await this.appConfigWriter.prependAppNameToModules(input.name);
            }
        }
        finally {
            process.chdir(cwd);
        }
        this.logger.info(ui_1.Text.create.taskCreateEnvironments);
        return result;
    }
}
exports.RegisterAppCommand = RegisterAppCommand;
