// @forge/cli-shared/out/app-logs/view-logs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViewAppLogsCommand = void 0;
const cs_ari_1 = require("@forge/util/packages/cs-ari");
const ari_1 = require("../ari");
class ViewAppLogsCommand {
    constructor(getAppConfig, appEnvironmentClient, globalEdgeClient, logsClient) {
        this.getAppConfig = getAppConfig;
        this.appEnvironmentClient = appEnvironmentClient;
        this.globalEdgeClient = globalEdgeClient;
        this.logsClient = logsClient;
    }
    async getAll({ environmentKey, limit, context, startTime = null }) {
        const { id: appId } = await this.getAppConfig();
        const [environmentId, cloudId] = await Promise.all([
            this.appEnvironmentClient.getAppEnvironmentId(appId, environmentKey),
            context ? this.globalEdgeClient.getCloudId(context.site) : undefined
        ]);
        const contextAri = cloudId ? (0, ari_1.buildContextAri)(cloudId, context.product) : undefined;
        const ari = (0, cs_ari_1.parse)(appId);
        const appAri = (0, cs_ari_1.parseAppAri)(ari);
        return this.logsClient.viewAppLogs({
            appId: appAri.appId,
            environmentId,
            limit,
            startTime,
            contextAri: contextAri ? [contextAri] : undefined
        });
    }
    async getOne(details) {
        return this.logsClient.viewInvocationLogs({ invocationId: details.invocationId });
    }
}
exports.ViewAppLogsCommand = ViewAppLogsCommand;
