// @forge/manifest/out/validators/product-trigger-scopes-validator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductTriggerScopesValidator = void 0;
const text_1 = require("../text");
const utils_1 = require("../utils");
const scopes_1 = require("../scopes");
class ProductTriggerScopesValidator {
    constructor(config) {
        this.config = config;
    }
    async validate(manifest) {
        var _a;
        if (!manifest || !manifest.typedContent) {
            return {
                success: false,
                manifestObject: manifest
            };
        }
        const validationErrors = [];
        const getRequiredScopes = (productEvent) => {
            var _a;
            const emptyRequiredScopes = { current: [] };
            return (((_a = this.config.mapping.find((value) => value.productEvent == productEvent)) === null || _a === void 0 ? void 0 : _a.oAuthScopes) || emptyRequiredScopes);
        };
        const addValidationError = (scope, event) => {
            validationErrors.push(Object.assign({ message: text_1.errors.permissions.missingPermissionFromScope(scope, event), reference: text_1.References.MissingScopes, level: 'error', metadata: {
                    missingPermission: scope
                } }, (0, utils_1.findPosition)('scopes', manifest.yamlContentByLine)));
        };
        if (!manifest.typedContent.modules || !manifest.typedContent.modules.trigger) {
            return {
                success: true,
                manifestObject: manifest
            };
        }
        const manifestScopes = ((_a = manifest.typedContent.permissions) === null || _a === void 0 ? void 0 : _a.scopes) || [];
        for (const element of manifest.typedContent.modules.trigger) {
            for (const event of element.events) {
                const requiredScopes = getRequiredScopes(event);
                const missingScopes = await (0, scopes_1.getMissingScopes)(manifestScopes, requiredScopes);
                missingScopes.forEach((scope) => addValidationError(scope, event));
            }
        }
        return {
            success: validationErrors.length === 0,
            manifestObject: manifest,
            errors: validationErrors
        };
    }
}
exports.ProductTriggerScopesValidator = ProductTriggerScopesValidator;
