// @forge/cli-shared/out/graphql/app-oauth-client-id-graphql-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppOauthClientGraphqlClient = void 0;
const index_1 = require("./index");
class AppOauthClientGraphqlClient {
    constructor(graphqlClient) {
        this.graphqlClient = graphqlClient;
    }
    async getAppOauthClientIdDetails(appAri, environmentKey) {
        const query = `
      query forge_cli_getAppOauthClientIdDetails($id: ID!, $key: String!) {
        app(id: $id) {
          name
          environmentByKey(key: $key) {
            id
            type
            oauthClient {
              clientID
            }
          }
        }
      }
    `;
        const result = await this.graphqlClient.query(query, {
            id: appAri,
            key: environmentKey
        });
        if (!result.app) {
            throw new index_1.MissingAppError();
        }
        if (!result.app.environmentByKey) {
            throw new index_1.MissingAppEnvironmentError(environmentKey);
        }
        return {
            name: result.app.name,
            appAri,
            environmentKey,
            environmentType: result.app.environmentByKey.type,
            environmentId: result.app.environmentByKey.id,
            oauthClientId: result.app.environmentByKey.oauthClient.clientID
        };
    }
}
exports.AppOauthClientGraphqlClient = AppOauthClientGraphqlClient;
