// @forge/cli-shared/out/graphql/gateway.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getEnvironment = exports.CDNEnvironments = exports.getGraphqlGateway = exports.DEFAULT_GRAPHQL_GATEWAY = void 0;
exports.DEFAULT_GRAPHQL_GATEWAY = 'https://api.atlassian.com/graphql';
const getGraphqlGateway = () => {
    return process.env.FORGE_GRAPHQL_GATEWAY || exports.DEFAULT_GRAPHQL_GATEWAY;
};
exports.getGraphqlGateway = getGraphqlGateway;
var Environments;
(function (Environments) {
    Environments["DEV"] = "dev";
    Environments["STG"] = "staging";
    Environments["PROD"] = "prod";
})(Environments || (Environments = {}));
var CDNEnvironments;
(function (CDNEnvironments) {
    CDNEnvironments["DEV"] = "dev";
    CDNEnvironments["STG"] = "stg";
    CDNEnvironments["PROD"] = "prod";
})(CDNEnvironments = exports.CDNEnvironments || (exports.CDNEnvironments = {}));
const getEnvironment = (envOptions = Environments) => {
    const graphqlGateway = (0, exports.getGraphqlGateway)();
    if (graphqlGateway.includes('.dev.')) {
        return envOptions.DEV;
    }
    if (graphqlGateway.includes('.stg.')) {
        return envOptions.STG;
    }
    return envOptions.PROD;
};
exports.getEnvironment = getEnvironment;
