// @forge/runtime/out/feature-flag.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeatureFlags = exports.XEN_RUNTIME_SHOULD_REPORT_METRICS = exports.XEN_RUNTIME_SHOULD_BLOCK_EXTERNAL_FETCH = exports.XEN_RUNTIME_ENABLE_NETWORK_LIMITS = exports.XEN_RUNTIME_EGRESS_DISCLOSURE_ENABLED = exports.XEN_RUNTIME_ENABLE_EGRESS_PROXY = exports.XEN_RUNTIME_ENABLE_EGRESS_FILTER = exports.SHOULD_ATTACH_TOKEN_FOR_APP = exports.SHOULD_ATTACH_TOKEN_FOR_ATLASSIAN_ACCOUNT = void 0;
exports.SHOULD_ATTACH_TOKEN_FOR_ATLASSIAN_ACCOUNT = 'should-attach-token-for-atlassian-account';
exports.SHOULD_ATTACH_TOKEN_FOR_APP = 'should-attach-token-for-app';
exports.XEN_RUNTIME_ENABLE_EGRESS_FILTER = 'xen-runtime-enable-egress-filter';
exports.XEN_RUNTIME_ENABLE_EGRESS_PROXY = 'xen-runtime-enable-egress-proxy';
exports.XEN_RUNTIME_EGRESS_DISCLOSURE_ENABLED = 'xen-runtime-egress-disclosure-enabled';
exports.XEN_RUNTIME_ENABLE_NETWORK_LIMITS = 'xen-runtime-enable-network-limits';
exports.XEN_RUNTIME_SHOULD_BLOCK_EXTERNAL_FETCH = 'xen-runtime-should-block-external-fetch';
exports.XEN_RUNTIME_SHOULD_REPORT_METRICS = 'xen-runtime-should-report-metrics';
class FeatureFlags {
    constructor(flags) {
        this.flags = flags;
    }
    static fromRequestMetaData(metaData) {
        return new FeatureFlags(metaData.featureFlags || []);
    }
    isFeatureFlagEnabled(flag) {
        return this.flags.includes(flag);
    }
}
exports.FeatureFlags = FeatureFlags;
