// @forge/manifest/out/types/module-types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SUPPORTED_PRODUCTS_ONLY = exports.SUPPORTED_MODULES = exports.AllModuleTypes = void 0;
var AllModuleTypes;
(function (AllModuleTypes) {
    AllModuleTypes["CoreFunction"] = "core:function";
    AllModuleTypes["CoreTrigger"] = "core:trigger";
    AllModuleTypes["CoreWebTrigger"] = "core:webtrigger";
    AllModuleTypes["CoreConsumer"] = "core:consumer";
    AllModuleTypes["CoreScheduledTrigger"] = "core:scheduledTrigger";
    AllModuleTypes["CoreEndpoint"] = "core:endpoint";
    AllModuleTypes["XenMacro"] = "xen:macro";
    AllModuleTypes["ConfluenceContentAction"] = "confluence:contentAction";
    AllModuleTypes["ConfluenceContentBylineItem"] = "confluence:contentBylineItem";
    AllModuleTypes["ConfluenceContextMenu"] = "confluence:contextMenu";
    AllModuleTypes["ConfluenceCustomContent"] = "confluence:customContent";
    AllModuleTypes["ConfluenceGlobalSettings"] = "confluence:globalSettings";
    AllModuleTypes["ConfluenceGlobalPage"] = "confluence:globalPage";
    AllModuleTypes["ConfluenceHomepageFeed"] = "confluence:homepageFeed";
    AllModuleTypes["ConfluenceSpacePage"] = "confluence:spacePage";
    AllModuleTypes["ConfluenceSpaceSettings"] = "confluence:spaceSettings";
    AllModuleTypes["JiraEntityProperty"] = "jira:entityProperty";
    AllModuleTypes["JiraWorkflowValidator"] = "jira:workflowValidator";
    AllModuleTypes["JiraWorkflowCondition"] = "jira:workflowCondition";
    AllModuleTypes["JiraWorkflowPostFunction"] = "jira:workflowPostFunction";
    AllModuleTypes["JiraCustomField"] = "jira:customField";
    AllModuleTypes["JiraCustomFieldType"] = "jira:customFieldType";
    AllModuleTypes["JiraIssueAction"] = "jira:issueAction";
    AllModuleTypes["JiraIssueActivity"] = "jira:issueActivity";
    AllModuleTypes["JiraUiModifications"] = "jira:uiModifications";
    AllModuleTypes["JiraIssueGlance"] = "jira:issueGlance";
    AllModuleTypes["JiraIssueContext"] = "jira:issueContext";
    AllModuleTypes["JiraIssuePanel"] = "jira:issuePanel";
    AllModuleTypes["JiraAdminPage"] = "jira:adminPage";
    AllModuleTypes["JiraProjectPage"] = "jira:projectPage";
    AllModuleTypes["JiraProjectSettingsPage"] = "jira:projectSettingsPage";
    AllModuleTypes["JiraGlobalPage"] = "jira:globalPage";
    AllModuleTypes["JiraDashboardGadget"] = "jira:dashboardGadget";
    AllModuleTypes["JiraDashboardBackgroundScript"] = "jira:dashboardBackgroundScript";
    AllModuleTypes["JiraViewIssueBackgroundScript"] = "jira:issueViewBackgroundScript";
    AllModuleTypes["JiraGlobalPermission"] = "jira:globalPermission";
    AllModuleTypes["JiraProjectPermission"] = "jira:projectPermission";
    AllModuleTypes["JiraJqlFunction"] = "jira:jqlFunction";
    AllModuleTypes["CompassAdminPage"] = "compass:adminPage";
    AllModuleTypes["CompassComponentPage"] = "compass:componentPage";
    AllModuleTypes["CompassGlobalPage"] = "compass:globalPage";
    AllModuleTypes["CompassTeamPage"] = "compass:teamPage";
    AllModuleTypes["CompassDataProvider"] = "compass:dataProvider";
    AllModuleTypes["JiraServiceManagementAssetsImportType"] = "jiraServiceManagement:assetsImportType";
    AllModuleTypes["JiraServiceManagementQueuePage"] = "jiraServiceManagement:queuePage";
    AllModuleTypes["JiraServiceManagementPortalRequestDetail"] = "jiraServiceManagement:portalRequestDetail";
    AllModuleTypes["JiraServiceManagementPortalRequestDetailPanel"] = "jiraServiceManagement:portalRequestDetailPanel";
    AllModuleTypes["JiraServiceManagementOrganizationPanel"] = "jiraServiceManagement:organizationPanel";
    AllModuleTypes["JiraServiceManagementPortalHeader"] = "jiraServiceManagement:portalHeader";
    AllModuleTypes["JiraServiceManagementPortalSubheader"] = "jiraServiceManagement:portalSubheader";
    AllModuleTypes["JiraServiceManagementPortalFooter"] = "jiraServiceManagement:portalFooter";
    AllModuleTypes["JiraServiceManagementPortalProfilePanel"] = "jiraServiceManagement:portalProfilePanel";
    AllModuleTypes["JiraServiceManagementPortalRequestViewAction"] = "jiraServiceManagement:portalRequestViewAction";
    AllModuleTypes["JiraServiceManagementPortalUserMenuAction"] = "jiraServiceManagement:portalUserMenuAction";
    AllModuleTypes["JiraServiceManagementPortalRequestCreatePropertyPanel"] = "jiraServiceManagement:portalRequestCreatePropertyPanel";
    AllModuleTypes["BitbucketRepoCodeOverviewCard"] = "bitbucket:repoCodeOverviewCard";
    AllModuleTypes["BitbucketRepoCodeOverviewAction"] = "bitbucket:repoCodeOverviewAction";
    AllModuleTypes["BitbucketRepoPullRequestCard"] = "bitbucket:repoPullRequestCard";
    AllModuleTypes["ConnectJiraAdminPages"] = "connect-jira:adminPages";
    AllModuleTypes["ConnectJiraJiraBackgroundScripts"] = "connect-jira:jiraBackgroundScripts";
    AllModuleTypes["ConnectJiraGeneralPages"] = "connect-jira:generalPages";
    AllModuleTypes["ConnectJiraJiraDashboardItems"] = "connect-jira:jiraDashboardItems";
    AllModuleTypes["ConnectJiraJiraGlobalPermissions"] = "connect-jira:jiraGlobalPermissions";
    AllModuleTypes["ConnectJiraJiraIssueFields"] = "connect-jira:jiraIssueFields";
    AllModuleTypes["ConnectJiraJiraIssueGlances"] = "connect-jira:jiraIssueGlances";
    AllModuleTypes["ConnectJiraJiraProjectAdminTabPanels"] = "connect-jira:jiraProjectAdminTabPanels";
    AllModuleTypes["ConnectJiraJiraProjectTabPanels"] = "connect-jira:jiraProjectTabPanels";
    AllModuleTypes["ConnectJiraJiraWorkflowConditions"] = "connect-jira:jiraWorkflowConditions";
    AllModuleTypes["ConnectJiraJiraWorkflowPostFunctions"] = "connect-jira:jiraWorkflowPostFunctions";
    AllModuleTypes["ConnectJiraKeyboardShortcuts"] = "connect-jira:keyboardShortcuts";
    AllModuleTypes["ConnectJiraWebItems"] = "connect-jira:webItems";
    AllModuleTypes["ConnectJiraLifecycle"] = "connect-jira:lifecycle";
    AllModuleTypes["ConnectConfluenceAdminPages"] = "connect-confluence:adminPages";
    AllModuleTypes["ConnectConfluenceBlueprints"] = "connect-confluence:blueprints";
    AllModuleTypes["ConnectConfluenceConfigurePage"] = "connect-confluence:configurePage";
    AllModuleTypes["ConnectConfluenceConfluenceContentProperties"] = "connect-confluence:confluenceContentProperties";
    AllModuleTypes["ConnectConfluenceConfluenceThemes"] = "connect-confluence:confluenceThemes";
    AllModuleTypes["ConnectConfluenceContentBylineItems"] = "connect-confluence:contentBylineItems";
    AllModuleTypes["ConnectConfluenceCustomContent"] = "connect-confluence:customContent";
    AllModuleTypes["ConnectConfluenceDialogs"] = "connect-confluence:dialogs";
    AllModuleTypes["ConnectConfluenceDynamicContentMacros"] = "connect-confluence:dynamicContentMacros";
    AllModuleTypes["ConnectConfluenceGeneralPages"] = "connect-confluence:generalPages";
    AllModuleTypes["ConnectConfluenceKeyboardShortcuts"] = "connect-confluence:keyboardShortcuts";
    AllModuleTypes["ConnectConfluencePostInstallPage"] = "connect-confluence:postInstallPage";
    AllModuleTypes["ConnectConfluenceSpaceToolsTabs"] = "connect-confluence:spaceToolsTabs";
    AllModuleTypes["ConnectConfluenceStaticContentMacros"] = "connect-confluence:staticContentMacros";
    AllModuleTypes["ConnectConfluenceWebItems"] = "connect-confluence:webItems";
    AllModuleTypes["ConnectConfluenceWebPanels"] = "connect-confluence:webPanels";
    AllModuleTypes["ConnectConfluenceWebSections"] = "connect-confluence:webSections";
    AllModuleTypes["ConnectConfluenceWebhooks"] = "connect-confluence:webhooks";
    AllModuleTypes["ConnectJiraConfigurePage"] = "connect-jira:configurePage";
    AllModuleTypes["ConnectJiraDialogs"] = "connect-jira:dialogs";
    AllModuleTypes["ConnectJiraJiraBuildInfoProvider"] = "connect-jira:jiraBuildInfoProvider";
    AllModuleTypes["ConnectJiraJiraDeploymentInfoProvider"] = "connect-jira:jiraDeploymentInfoProvider";
    AllModuleTypes["ConnectJiraJiraDevelopmentTool"] = "connect-jira:jiraDevelopmentTool";
    AllModuleTypes["ConnectJiraJiraEntityProperties"] = "connect-jira:jiraEntityProperties";
    AllModuleTypes["ConnectJiraJiraFeatureFlagInfoProvider"] = "connect-jira:jiraFeatureFlagInfoProvider";
    AllModuleTypes["ConnectJiraJiraIssueContents"] = "connect-jira:jiraIssueContents";
    AllModuleTypes["ConnectJiraJiraIssueTabPanels"] = "connect-jira:jiraIssueTabPanels";
    AllModuleTypes["ConnectJiraJiraIssueTypes"] = "connect-jira:jiraIssueTypes";
    AllModuleTypes["ConnectJiraJiraProjectPages"] = "connect-jira:jiraProjectPages";
    AllModuleTypes["ConnectJiraJiraProjectPermissions"] = "connect-jira:jiraProjectPermissions";
    AllModuleTypes["ConnectJiraJiraReports"] = "connect-jira:jiraReports";
    AllModuleTypes["ConnectJiraJiraSearchRequestViews"] = "connect-jira:jiraSearchRequestViews";
    AllModuleTypes["ConnectJiraJiraTimeTrackingProviders"] = "connect-jira:jiraTimeTrackingProviders";
    AllModuleTypes["ConnectJiraJiraWorkflowValidators"] = "connect-jira:jiraWorkflowValidators";
    AllModuleTypes["ConnectJiraPostInstallPage"] = "connect-jira:postInstallPage";
    AllModuleTypes["ConnectJiraWebPanels"] = "connect-jira:webPanels";
    AllModuleTypes["ConnectJiraWebSections"] = "connect-jira:webSections";
    AllModuleTypes["ConnectJiraWebhooks"] = "connect-jira:webhooks";
    AllModuleTypes["ConnectConfluenceLifecycle"] = "connect-confluence:lifecycle";
    AllModuleTypes["AutomationForConfluenceTrigger"] = "automation:trigger";
    AllModuleTypes["AutomationForConfluenceAction"] = "automation:action";
    AllModuleTypes["AutomationForConfluenceBranch"] = "automation:branch";
    AllModuleTypes["AutomationForConfluenceCondition"] = "automation:condition";
    AllModuleTypes["AutomationForConfluenceAttributeResolver"] = "automation:attributeResolver";
    AllModuleTypes["AutomationForConfluenceTemplate"] = "automation:template";
    AllModuleTypes["AutomationForConfluenceI18n"] = "automation:i18n";
})(AllModuleTypes = exports.AllModuleTypes || (exports.AllModuleTypes = {}));
exports.SUPPORTED_MODULES = Object.values(AllModuleTypes);
exports.SUPPORTED_PRODUCTS_ONLY = [...new Set(exports.SUPPORTED_MODULES.map((module) => module.split(':')[0]))];
