// @forge/cli-shared/out/shared/product.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isSupportedProduct = exports.isBitbucketProduct = exports.isWorkspaceProduct = exports.capitalizeProduct = exports.WORKSPACE_PRODUCTS = exports.SUPPORTED_PRODUCTS = void 0;
const case_1 = require("case");
exports.SUPPORTED_PRODUCTS = ['Jira', 'Confluence', 'Compass'];
exports.WORKSPACE_PRODUCTS = ['Bitbucket'];
function capitalizeProduct(product) {
    return (0, case_1.capital)(product);
}
exports.capitalizeProduct = capitalizeProduct;
const isWorkspaceProduct = (product) => {
    return exports.WORKSPACE_PRODUCTS.includes(capitalizeProduct(product));
};
exports.isWorkspaceProduct = isWorkspaceProduct;
const isBitbucketProduct = (product) => {
    return capitalizeProduct(product) === 'Bitbucket';
};
exports.isBitbucketProduct = isBitbucketProduct;
const isSupportedProduct = (product) => {
    return exports.SUPPORTED_PRODUCTS.includes(capitalizeProduct(product)) || (0, exports.isWorkspaceProduct)(product);
};
exports.isSupportedProduct = isSupportedProduct;
